signac/contrib/collection.py

```diff
@@ -493,6 +494,24 @@ def __getitem__(self, _id):
             self._assert_open()
             raise
 
+    @staticmethod
+    def _validate_key(key):
+        "Emit a warning or raise an exception if key is invalid. Returns key."
+        if '.' in key:
+            from ..errors import InvalidKeyError
```

Comment:
We normally don't put imports nested anywhere unless it's a `try...except` block, is there a reason not to put this at the top of the file?

Reply:
Yes, because there are otherwise module interdependencies.